### PR TITLE
[INJIMOB-903] clean local backup files post backup / restore

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
Other changes
- Fetch last backup details from cloud (GDrive / iCloud) everytime user presses on backup & restore option in settings
- In case of no internet during backup / restore in iOS, show no network error toaster

### CODE REVIEW

File: screens/backupAndRestore/BackupAndRestoreSetupController.ts [hunk 1]
```diff
@@ -33,10 +38,27 @@ export function useBackupAndRestoreSetup() {
     selectIsBackUpAndRestoreExplored,
   );
 
+  const isNetworkErrorDuringAccountSetup = useSelector(
+    service,
+    selectIsNetworkOff,
```
Comment (Alka1703) [inline screens/backupAndRestore/BackupAndRestoreSetupController.ts:43]: change it to networkError to make it consistent

File: screens/backupAndRestore/BackupAndRestoreSetupController.ts [hunk 2]
```diff
@@ -61,9 +83,15 @@ export function useBackupAndRestoreSetup() {
...
-    TRY_AGAIN: () => service.send(BackupAndRestoreSetupEvents.TRY_AGAIN()),
+    RECONFIGURE_ACCOUNT: () =>
+      service.send(BackupAndRestoreSetupEvents.RECONFIGURE_ACCOUNT()),
+    TRY_AGAIN: tryAgain,
     OPEN_SETTINGS: () =>
       service.send(BackupAndRestoreSetupEvents.OPEN_SETTINGS()),
-    DISMISS: () => service.send(BackupAndRestoreSetupEvents.DISMISS()),
+    DISMISS: () => {
+      if (isNetworkErrorWhileFetchingLastBackupDetails)
+        backupService?.send(BackupAndRestoreSetupEvents.DISMISS());
```
Comment (Alka1703) [inline screens/backupAndRestore/BackupAndRestoreSetupController.ts:93]: update it to senrd eent to backup machine instead of setup machine

**Review by Alka1703 — CHANGES_REQUESTED** [review]
Please make necessary changes

File: machines/backupAndRestore/backup.ts [hunk 3]
```diff
@@ -65,62 +68,111 @@ export const backupMachine = model.createMachine(
...
-              STORE_RESPONSE: [
+          idle: {},
+          checkCloud: {
+            invoke: {
+              src: 'getLastBackupDetailsFromCloud',
+              onDone: {
+                actions: [
+                  'unsetIsLoadingBackupDetails',
+                  'setLastBackupDetails',
+                  'sendBackupAndRestoreSetupSuccessEvent',
```
[the remark this answers is not in the document]
Comment (KiruthikaJeyashankar) [reply machines/backupAndRestore/backup.ts:90]: Done

File: machines/backupAndRestore/backup.ts [hunk 4]
```diff
@@ -271,24 +326,16 @@ export const backupMachine = model.createMachine(
...
-          },
-        },
       },
     },
   },
   {
     actions: {
-      unsetIsLoading: model.assign({
-        isLoading: false,
+      unsetIsLoadingBackupDetails: model.assign({
```
Comment (vharsh) [inline machines/backupAndRestore/backup.ts:334]: (doubt) This action manages the green/red status bar on the top part of the app which doesn't show up on silent backups, am I right here? I was wondering why this isn't being called on silentFailures due to lack of internet.

File: machines/backupAndRestore/backupAndRestoreSetup.ts [hunk 5]
```diff
@@ -330,30 +344,24 @@ export const backupAndRestoreSetupMachine = model.createMachine(
...
     guards: {
-      isInternetConnected: (_, event) => event.data.isConnected,
+      isInternetConnected: (_, event) =>
+        !!(event.data as NetInfoState).isConnected,
       isNetworkError: (_, event) => event.data.error === NETWORK_REQUEST_FAILED,
       isSignedIn: (_context, event) =>
         (event.data as isSignedInResult).isSignedIn,
-      isNotSignedInIOS: (_context, event) => {
+      isIOSAndNotSignedIn: (_context, event) => {
         return !(event.data as isSignedInResult).isSignedIn && isIOS();
```
Comment (vharsh) [inline machines/backupAndRestore/backupAndRestoreSetup.ts:363]: Please invert the order of evaluation as on Android the isSignedInResult has a different structure iirc.

File: machines/backupAndRestore/backup.ts (hunk 4 above)
Comment (KiruthikaJeyashankar) [reply machines/backupAndRestore/backup.ts:334]: No, this is handling fetching of last backup details

**Review by vharsh — APPROVED** [review]
lgtm

**Review by swatigoel — APPROVED** [review]

**Pull request merged**